server/data/data-library/src/main/java/org/apache/james/user/lib/AbstractUsersRepository.java

```diff
@@ -20,36 +20,44 @@
 package org.apache.james.user.lib;
 
 import java.util.Optional;
+import java.util.stream.Stream;
+
 import javax.inject.Inject;
 
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.apache.james.core.MailAddress;
 import org.apache.james.domainlist.api.DomainList;
-import org.apache.james.domainlist.api.DomainListException;
 import org.apache.james.lifecycle.api.Configurable;
 import org.apache.james.user.api.AlreadyExistInUsersRepositoryException;
 import org.apache.james.user.api.UsersRepository;
 import org.apache.james.user.api.UsersRepositoryException;
-import org.apache.commons.configuration.ConfigurationException;
-import org.apache.commons.configuration.HierarchicalConfiguration;
-import org.apache.james.core.MailAddress;
+import org.apache.james.util.OptionalUtils;
 
+import com.github.steveash.guavate.Guavate;
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
 
 public abstract class AbstractUsersRepository implements UsersRepository, Configurable {
-
-    private DomainList domainList;
+    public static final String ENABLE_VIRTUAL_HOSTING = "enableVirtualHosting";
+    public static final String ADMINISTRATOR_ID = "administratorId";
     private boolean virtualHosting;
     private Optional<String> administratorId;
+    private DomainList domainList;
+    protected UsernameValidator usernameValidator = new UsernameValidatorAggregator(ImmutableSet.of());
```

Comment:
you increased complexity of the code by sharing more things in the hierarchy. I would rather extract things to another class to keep responsibility as small as possible here.